fastparquet/writer.py

```diff
@@ -571,6 +574,12 @@ def write(filename, data, partitions=[0], encoding="PLAIN",
                     fmd.row_groups.append(rg)
             fmd.num_rows = sum(rg.num_rows for rg in fmd.row_groups)
         else:
+            if write_index or write_index is None and not (
```

Comment:
Well, I really want to keep the partitioning, and enable it for dask-dataframe too - it's a big win in terms of space-saving and avoiding reading some partitions, if people use it. I don't immediately see where else it could go. Of course, if we instead accept "partition columns" with a dataframe and not a groupby object, we can apply the reset index up front. It'll still need some repetition in the dask writer.